packages/web/src/components/swap/select-token-modal/SelectTokenModal.tsx

```diff
@@ -0,0 +1,104 @@
+// TODO : remove eslint-disable after work
+/* eslint-disable */
+import { Token } from "@containers/header-container/HeaderContainer";
+import React, { useEffect, useRef } from "react";
+import {
+  SearchModalBackground,
+  SearchModal,
+  InputStyle,
+} from "./SelectTokenModal.styles";
+import IconSearch from "@components/common/icons/IconSearch";
+import IconClose from "@components/common/icons/IconCancel";
+import { coinListProps } from "@containers/swap-container/SwapContainer";
+
+interface SelectTokenModalProps {
+  onSelectTokenModal: () => void;
+  search: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  keyword: string;
+  coinList: coinListProps[];
+  changeToken: (token: coinListProps, type: string) => void;
+}
+
+const SelectTokenModal: React.FC<SelectTokenModalProps> = ({
+  onSelectTokenModal,
+  search,
+  keyword,
+  coinList,
+  changeToken,
+}) => {
+  const menuRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    const closeMenu = (e: MouseEvent) => {
+      if (menuRef.current && menuRef.current.contains(e.target as Node)) {
+        return;
+      } else {
+        e.stopPropagation();
+        onSelectTokenModal();
+      }
+    };
+    window.addEventListener("click", closeMenu, true);
+    return () => {
+      window.removeEventListener("click", closeMenu, true);
+    };
+  }, [menuRef, onSelectTokenModal]);
+
+  return (
+    <SearchModalBackground>
+      <SearchModal ref={menuRef}>
+        <div className="modal-body">
+          <div className="modal-header">
+            <span>Select a token</span>
+            <div className="close-wrap" onClick={onSelectTokenModal}>
+              <IconClose className="close-icon" />
+            </div>
+          </div>
+          <div className="search-wrap">
+            <InputStyle
+              placeholder={"Search name or paste address"}
+              value={keyword}
+              onChange={search}
```

Comment:
It seems to be missing the ability to retrieve a list of tokens by the `search` function.

Reply:
good ! added.